### DISCUSSION TITLE
What is the usage of these lora methods ?

### DISCUSSION THREAD

**Discussion by hamzaakyildiz** [huggingface/diffusers#9117] [Q&A] [answered]
I'm currently working with LoRA weights and their combinations, but I'm not entirely sure if I fully understand their usage. Could you help clarify a few points?

    Loading/Unloading LoRA Weights
    Fusing/Unfusing LoRAs
    Setting Adapters

Questions:

    Optimal Combination for Using Two LoRAs:
        When I want to use two LoRA weights simultaneously, what is the optimal way to combine them?

    Setting Different Scales for LoRAs:
        How can I set different scales for my LoRAs to ensure they work as intended?

    Sequence for Unloading and Loading LoRAs:
        If I need to unload the LoRAs used in step 2 and load another two LoRAs, what sequence of code should I follow to achieve this?

Any insights or examples would be greatly appreciated!

**Comment by asomoza** [comment, marked as answer]
Hi, all that you're asking about usage is in the docs with examples:

- [Loading/Unloading LoRA Weights](https://huggingface.co/docs/diffusers/tutorials/using_peft_for_inference)
- [Fusing/Unfusing LoRAs](https://huggingface.co/docs/diffusers/using-diffusers/merge_loras#fuselora)
- [Setting Adapters](https://huggingface.co/docs/diffusers/using-diffusers/merge_loras#setadapters)

About your questions:

>  When I want to use two LoRA weights simultaneously, what is the optimal way to combine them?

There's no universal solution for this. It depends a lot on how they were trained, most really good loras trained from experienced trainers don't need anything special to be used in conjunction with other loras if they don't clash in the concepts. But it is always better if you load them with different scales and different blocks if needed, but this is something you'll have to experiment with.

> How can I set different scales for my LoRAs to ensure they work as intended?

How to set different scales is in the documentation, how to ensure they work as intended doesn't depend on diffusers but the lora training. Overfitted loras need lower scales which is bad, undertrained loras need bigger scales which is also bad, ideally a good lora to work as intended with good training should works with a scale of 1.0

> If I need to unload the LoRAs used in step 2 and load another two LoRAs, what sequence of code should I follow to achieve this?

This is pretty straightforward, just load them, unload them and load the new ones. To unload them you'll need to use:

```python
pipeline.unload_lora_weights()
```

**Comment by hamzaakyildiz** [comment]
Thank you for the help
